pkg/daemonsetlock/daemonsetlock.go

```diff
@@ -137,6 +144,24 @@ func (dsl *DaemonSetLock) Release() error {
 	}
 }
 
+// GetDaemonSet returns the named DaemonSet resource from the DaemonSetLock's configured client
+func (dsl *DaemonSetLock) GetDaemonSet(sleep, timeout time.Duration) (*v1.DaemonSet, error) {
```

Comment:
You pass duration parameters, but use only the consts in the method body -- pick one or other approach (I would remove the parameters).

Reply:
oh, that's an oversight!

I think the method should take the args so we don't refer to global vars in the business logic itself, and to allow future re-use using a different sleep/timeout config. Will fix now...